lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1455,29 +1433,47 @@ void LocalEnforcer::init_policy_reauth_for_session(
       to_vec(request.dynamic_rules_to_install()), rules_to_activate,
       rules_to_deactivate, update_criteria);
 
-  auto ip_addr = session->get_config().common_context.ue_ipv4();
-  if (rules_to_process_is_not_empty(rules_to_deactivate)) {
-    deactivate_success = pipelined_client_->deactivate_flows_for_rules(
-        request.imsi(), rules_to_deactivate.static_rules,
-        rules_to_deactivate.dynamic_rules, RequestOriginType::GX);
-  }
-  if (rules_to_process_is_not_empty(rules_to_activate)) {
-    activate_success = pipelined_client_->activate_flows_for_rules(
-        request.imsi(), ip_addr, rules_to_activate.static_rules,
-        rules_to_activate.dynamic_rules,
-        std::bind(
-            &LocalEnforcer::handle_activate_ue_flows_callback, this,
-            request.imsi(), ip_addr, rules_to_activate.static_rules,
-            rules_to_activate.dynamic_rules, _1, _2));
-  }
+  propagate_rule_updates_to_pipelined(
+      imsi, session->get_config(), rules_to_activate, rules_to_deactivate,
+      false);
 
   if (terminate_on_wallet_exhaust() && is_wallet_exhausted(*session)) {
     start_session_termination(imsi, session, true, update_criteria);
     return;
   }
-  if (!session->is_radius_cwf_session()) {
+  if (session->get_config().common_context.rat_type() == TGPP_LTE) {
     create_bearer(
         activate_success, session, request, rules_to_activate.dynamic_rules);
+    // TODO propagate RaR triggered bearer updates to rule based
+    // const auto update = session->get_dedicated_bearer_updates(
+    //    rules_to_activate, rules_to_deactivate, update_criteria);
+    // propagate_bearer_updates_to_mme(update);
+  }
```

Comment:
@ulaskozat Should we be able to migrate this existing `create_bearer` call to the rule based approach? Is the QoS profile coming in from the RAR attached to a rule? It looks like there could be multiple rules sent as part of the same request... Does the same QoS profile apply to all of them?

Reply:
each policy rule has its own QoS profile. when you issue create bearer request to MME, SPGW task parses each policy rule in that request and creates distinct bearer requests.